lib/build-cloud/securitygroup.rb

```diff
@@ -80,21 +80,35 @@ def rationalise_rules( authorized_ranges )
         
         security_group.ip_permissions.each do |r|
 
-            c = {
-                :min_port    => r['fromPort'],
-                :max_port    => r['toPort'],
-                :ip_protocol => r['ipProtocol'],
-            }
-
             if r['groups'] != []
```

Comment:
Hmm.. on second thought, `if r['groups'] != []` != `unless r['groups'].empty?` - it doesn't take into consideration the possibility of r['groups'] not being set as an array